src/nvim/regexp.c

```diff
@@ -6782,19 +6750,20 @@ static int vim_regsub_both(char_u *source, typval_T *expr, char_u *dest,
           }
             c = *src++;
           }
-        } else if (has_mbyte)
-          c = mb_ptr2char(src - 1);
-
-        /* Write to buffer, if copy is set. */
-        if (func_one != (fptr_T)NULL)
-          /* Turbo C complains without the typecast */
+        } else {
+          c = utf_ptr2char(src - 1);
+        }
+        // Write to buffer, if copy is set.
+        if (func_one != NULL) {
           func_one = (fptr_T)(func_one(&cc, c));
-        else if (func_all != (fptr_T)NULL)
-          /* Turbo C complains without the typecast */
-          func_all = (fptr_T)(func_all(&cc, c));
-        else     /* just copy */
-          cc = c;
-
+        } else {
```

Comment:
Why was this changed into a plain `} else {` rather than kept as the `} else if (func_all != NULL) {`?

Reply:
@ZviRackover Linter is completely fine with the form @jamessan suggested. You need to fix linter errors, but there is no reason to use `else { if }` form in place of `else if`, you just needed to add braces after the `if (cond)`.

Reply:
Done in [4f3ce52](https://github.com/neovim/neovim/pull/8665/commits/4f3ce526e644628b200257b4bcf867eb438f8c12])